### PR TITLE
Issue #260 - Handle invalid or expired session keys to Hydrus

### PR DESCRIPTION
**Pull request by professor-mesmerize**
For the Hydrus source, this PR switches FlipFlip to exclusively use the API Access Key instead of a mix of the API Access Key and the Session Key. We were already using the API Access Key on every file fetch anyways, so it's not a significant change in security, and the API Access Key doesn't have a time-based expiration.

Upon testing locally, this fix seems to resolves Issue #260, where FlipFlip would hold on to an expired session key without fetching a new one. (However, it's somewhat difficult to test thoroughly, as that expired condition isn't the easiest to simulate artificially.)

### CODE REVIEW

File: src/renderer/components/player/SourceScraper.tsx [hunk 1]
```diff
@@ -1529,12 +1509,12 @@ function loadHydrus(systemMessage: Function, config: Config, source: LibrarySour
           .catch((e) => {
             resolve(null);
           });
-      }
+      };
```
Comment (professor-mesmerize) [inline src/renderer/components/player/SourceScraper.tsx:1512]: @StarrHelixx There were a few semi-colons that seemed to be missing here and there in this part of the code? But if they shouldn't have been added, I'm happy to remove them!

File: src/renderer/components/player/SourceScraper.tsx [hunk 2]
```diff
@@ -1563,10 +1543,10 @@ function loadHydrus(systemMessage: Function, config: Config, source: LibrarySour
...
 
-      if (sessionKey == null) {
-        getSessionKey();
+      if (apiKey == null) {
+        apiKey = config.remoteSettings.hydrusAPIKey;
       } else {
         search();
       }
```
Comment (professor-mesmerize) [inline src/renderer/components/player/SourceScraper.tsx:1552]: @StarrHelixx I think that this if-else block could potentially just be removed and replaced with a simple call to `search();`, but I wasn't quite confident enough in variable scoping in TypeScript to be sure so I left it in. If you'd rather I remove it and just call `search()`, though, let me know and I'll update it!

Comment (StarrHelixx) [reply src/renderer/components/player/SourceScraper.tsx:1552]: Yup, no need to call this block, just `search()`

**Comment by StarrHelixx** [comment]
This has been moved to #273

**Comment by professor-mesmerize** [comment]
@StarrHelixx Thank you! I have no idea where my branches got the commits for PRs 1 and 2, that's wild. 😂 I'll go ahead and test all of these out!

**Pull request closed without merging**